tensorflow_addons/image/transform_test.py

```diff
@@ -141,6 +147,156 @@ def test_transform_eager(self):
+                      [2, 7, 8, 13, 19],
+                      [1, 6, 12, 18, 23],
+                      [5, 11, 16, 17, 22],
+                      [0, 10, 15, 21, 0]],
+                     [[20, 15, 10, 5, 0],
+                      [21, 16, 11, 6, 1],
+                      [22, 17, 12, 7, 2],
+                      [23, 18, 13, 8, 3],
+                      [24, 19, 14, 9, 4]]])
+                # yapf: enable
+
+    def test_compose_rotate(self):
+        for dtype in _DTYPES:
+            with self.cached_session():
+                image = constant_op.constant(
+                    [[1, 1, 1, 0], [1, 0, 0, 0], [1, 1, 1, 0], [0, 0, 0, 0]],
+                    dtype=dtype)
+                # Rotate counter-clockwise by pi / 2.
+                rotation = transform_ops.angles_to_projective_transforms(
+                    np.pi / 2, 4, 4)
+                # Translate right by 1 (the transformation matrix is always inverted,
+                # hence the -1).
+                translation = constant_op.constant([1, 0, -1, 0, 1, 0, 0, 0],
+                                                   dtype=dtypes.float32)
+                composed = transform_ops.compose_transforms(
+                    rotation, translation)
+                image_transformed = transform_ops.transform(image, composed)
+                self.assertAllEqual(
+                    image_transformed,
+                    [[0, 0, 0, 0], [0, 1, 0, 1], [0, 1, 0, 1], [0, 1, 1, 1]])
+
+    def test_bilinear(self):
+        with self.cached_session():
+            image = constant_op.constant(
+                [[0, 0, 0, 0, 0], [0, 1, 1, 1, 0], [0, 1, 0, 1, 0],
+                 [0, 1, 1, 1, 0], [0, 0, 0, 0, 0]], dtypes.float32)
+            # The following result matches:
+            # >>> scipy.ndimage.rotate(image, 45, order=1, reshape=False)
+            # which uses spline interpolation of order 1, equivalent to bilinear
+            # interpolation.
+            self.assertAllClose(
+                transform_ops.rotate(
+                    image, np.pi / 4.0, interpolation="BILINEAR"),
+                [[0.000, 0.000, 0.343, 0.000, 0.000],
+                 [0.000, 0.586, 0.914, 0.586, 0.000],
+                 [0.343, 0.914, 0.000, 0.914, 0.343],
+                 [0.000, 0.586, 0.914, 0.586, 0.000],
+                 [0.000, 0.000, 0.343, 0.000, 0.000]],
+                atol=0.001)
+            # yapf: disable
+            self.assertAllClose(
+                transform_ops.rotate(
+                    image, np.pi / 4.0, interpolation="NEAREST"),
+                [[0, 0, 1, 0, 0],
+                 [0, 1, 1, 1, 0],
+                 [1, 1, 0, 1, 1],
+                 [0, 1, 1, 1, 0],
+                 [0, 0, 1, 0, 0]])
+            # yapf: enable
+
+    def test_bilinear_uint8(self):
+        with self.cached_session():
+            image = constant_op.constant(
+                np.asarray(
+                    [[0.0, 0.0, 0.0, 0.0, 0.0], [0.0, 255, 255, 255, 0.0],
```

Comment:
Perhaps it's always better to format those arrays manually :-)

Reply:
done